#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_rviz_plugins'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("width", int_t, 0, "", 1200, -1, 2024)
gen.add("height", int_t, 0, "", 800, -1, 2024)
gen.add("top", int_t, 0, "", 10, 0, 2024)
gen.add("left", int_t, 0, "", 10, 0, 2024)

gen.add("text_size", double_t, 0, "", 12, 1, 32)

gen.add("bg_red", double_t, 0, "", 0, 0, 1.0)
gen.add("bg_blue", double_t, 0, "", 0, 0, 1.0)
gen.add("bg_green", double_t, 0, "", 0, 0, 1.0)
gen.add("bg_alpha", double_t, 0, "", 0, 0, 1.0)

gen.add("fg_red", double_t, 0, "", 25.0 / 255, 0, 1.0)
gen.add("fg_blue", double_t, 0, "", 1.0, 0, 1.0)
gen.add("fg_green", double_t, 0, "", 240/255.5, 0, 1.0)
gen.add("fg_alpha", double_t, 0, "", 1, 0, 1.0)

exit (gen.generate (PACKAGE, "jsk_rviz_plugins", "OverlayTextInterface"))
